src/open_samus_returns_rando/files/custom/scenario.lua

```diff
@@ -201,6 +201,39 @@ function Scenario.RandoOnElevatorUse(from_actor, _ARG_1_, _ARG_2_)
   Elevator.Use("c10_samus", destination.scenario, destination.actor, _ARG_2_)
 end
 
+function Scenario.ShowFinalBossMessage()
+  if Init.sFinalBoss == "Ridley" then return end
+  local boss = Init.sFinalBoss
+  local startpoint = boss
+  if boss == "Diggernaut" then
+    startpoint = "ManicMiner"
+  elseif boss == "Queen" then
+    boss = "the Queen"
+  end
+  GUI.LaunchMessage("Not enough Metroid DNA!\nCollect more DNA to fight " .. boss .. "!", "RandomizerPowerup.Dummy", "")
+  if Init.sFinalBoss ~= "Queen" then
+    Game.AddSF(0, "Scenario.FinalBossReload", "s", startpoint)
+  end
+end
+
+function Scenario.FinalBossReload(startpoint)
+  Scenario.LoadNewScenario(Scenario.CurrentScenarioID, "ST_SG_" .. startpoint)
+end
+
+function Scenario.LaunchCredits()
+  Game.ShowEndGameCredits(true)
+end
+
+function Scenario.OnPostCreditsEnd()
+  Game.SaveGameComplete()
+  if CurrentScenario.bFirstTimeCompleted then
+    Game.StopEnvironmentSound()
+    Game.AddGUISF(2.5, GUI.MainMenuGoToState, "i", 28)
+  else
```

Comment:
Does this have any use now? Before the `bFirstTimeCompleted` was added to every scenario but was always false and now it doesn't even exist besides the vanilla thing in s110_surfaceb.

Reply:
Honestly I'm not even sure. It seems to always just do the else regardless. I even deleted all Save data and hard was just unlocked on the file and completed after finishing, but had no popup. Probably could be removed then

Reply:
Even when the scenarios had those variables it still was never called so I'm not sure what it's for. I think Ridley is handled differently.

Reply:
Tested without it and it still finished just fine, so I'm removing it. Hopefully that's everything?